### ISSUE TITLE
add success animation

### ISSUE THREAD

**Issue by rolznz** [getAlby/bitcoin-connect#100]
Shown on connect / successful payment

![image](https://github.com/getAlby/bitcoin-connect/assets/33993199/3739349e-33f3-4f54-a3df-d88f10a2a611)
![New Animation](https://github.com/getAlby/bitcoin-connect/assets/33993199/e0589f0d-eee3-40d0-a1de-c9a9b83695bb)
[New Animation (Lottie).zip](https://github.com/getAlby/bitcoin-connect/files/13316812/New.Animation.Lottie.zip)
[New Animation (gif).zip](https://github.com/getAlby/bitcoin-connect/files/13316820/New.Animation.gif.zip)

I wonder if Lottie is going to be too heavy for our library. WHat if we just use the gif?

**Comment by rolznz** [comment]
Currently using the gif encoded as base64, but the filesize is large and the quality is poor

**Comment by rolznz** [comment]
This still needs to be added to connect (currently only shown after payment)

**Comment by rolznz** [comment]
As per https://github.com/getAlby/bitcoin-connect/issues/131#issuecomment-1832941072 the success animation should show and then the modal should automatically close.

**Comment by Nomishka** [comment]
Hey @rolznz, The thing should be solved in this issue is, Close this bc-modal automatically when the wallet connected right?

![image](https://github.com/user-attachments/assets/2f0b2f90-3aaf-4447-b9fd-87b722d27cda)

**Comment by rolznz** [comment]
@Nomishka yes that's right. But before closing, show the success animation first (as per the screenshot)

**Comment by Nomishka** [comment]
@rolznz, Is this issue allowed as a bounty?

**Comment by Nomishka** [comment]
@rolznz, Thanks! I hope to continue contributing.

**Comment by Nomishka** [comment]
Hey @rolznz, I successfully added the success animation and automatically closing logic for the modal. But I have a question about it. When I click again on the bc-button which shows "connected", the modal opens again and it triggers the animation again. I guess it doesn't need to trigger the animation again as it is already connected. What is your expectation about it?

[Screencast from 2025-01-15 22-43-48.webm](https://github.com/user-attachments/assets/d712d8c2-ce53-4de2-b628-58c16fc320b2)

**Comment by rolznz** [comment]
@Nomishka it should not trigger the animation when you open the modal again. It should show the actual modal content. (The balance, disconnect button etc.)

**Comment by Nomishka** [comment]
@rolznz got it. Thanks!

**Comment by Nomishka** [comment]
> [@Nomishka](https://github.com/Nomishka) it should not trigger the animation when you open the modal again. It should show the actual modal content. (The balance, disconnect button etc.)

@rolznz, Is this good to solve in the same PR which solves this issue or in seperate PR?

**Comment by rolznz** [comment]
@Nomishka it needs to be done in the same PR. Can you see why? what happens if it always shows the success animation and closes when you try to open the modal?
